### PR TITLE
Fill vignette titles in YAML with `use_vignette()`.

### PR DESCRIPTION
**Pull request by rorynolan**
Hi Jenny,
Here's the PR from https://github.com/r-lib/usethis/issues/445.
I'm pretty sure it works, I've tested it manually. Before I write a unit test for it, I'd like you to advise me how to. My instinct is to put a file in `inst/for-testthat` (or something like that) that should be identical to an output of `use_vignette("Cool thing")`, but that's probably not the way you do things.
Rory

### CODE REVIEW

**Comment by jennybc** [comment]
I'm working on something else now FYI, not ignoring you. usethis just had a release, so it will lie fallow for a bit.

**Comment by rorynolan** [comment]
No worries, I realise this would be a low-priority issue even if you were working on `usethis`. I might tinker around a bit with this PR, but you can safely ignore me until you're ready to deal with this.

File: R/vignette.R [hunk 1]
```diff
@@ -34,9 +34,13 @@ use_vignette <- function(name) {
   path <- proj_path("vignettes", asciify(name), ext = "Rmd")
 
   done("Creating {value(proj_rel_path(path))}")
-  rmarkdown::draft(
-    path, "html_vignette", "rmarkdown",
-    create_dir = FALSE, edit = FALSE
+  use_template("vignette.Rmd",
+    save_as = proj_rel_path(path),
+    data = list(
+      vignette_title = name,
```
Comment (hadley) [inline R/vignette.R]: I think it would be better to add a new argument, `title`.

File: inst/templates/vignette.Rmd [hunk 2]
```diff
@@ -0,0 +1,65 @@
+---
+title: "{{{ vignette_title }}}"
+author: "{{{ vignette_author }}}"
```
Comment (hadley) [inline inst/templates/vignette.Rmd]: I think we should just remove the `author` and `date` fields altogether. I don't think they're very useful.

**Comment by hadley** [comment]
@rorynolan are you interested in finishing this off? No problems if not, just let me know so I can get it over the finish line.

**Comment by rorynolan** [comment]
Sure @hadley,
Ill do this tomorrow. 
Rory

**Comment by rorynolan** [comment]
OK @hadley I think this is good to go.

**Comment by rorynolan** [comment]
Alrighty @hadley, I've done that now :-)

**Comment by hadley** [comment]
Thanks @rorynolan! 🎉

**Pull request merged**